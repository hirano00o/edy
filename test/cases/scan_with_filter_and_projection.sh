#!/bin/bash

SCRIPT_ROOT_DIR=$1
TEST_NAME=$(basename "$0" | sed "s/\..*//")

# aws dynamodb scan --table-name User \
#   --projection-expression "Email,Address.City"
#   --filter-expression "contains(Interest.SNS,:sns) and attribute_exists(Interest.Video)" \
#   --expression-attribute-values "{\":sns\":{\"S\":\"Twitter\"}}" \
#   --endpoint-url http://localhost:8000
CMD="edy s -t User -f \"Interest.SNS,S contains Twitter and Interest.Video,SS exists\" \
  --pj \"Email Address.City\" --local 8000"

. "${SCRIPT_ROOT_DIR}"/helper.sh

run_such_query_helper