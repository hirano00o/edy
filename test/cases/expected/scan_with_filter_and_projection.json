[
  {
    "Address": {
      "City": "Dover"
    },
    "Email": "dave@example.com"
  },
  {
    "Email": "frank@example.com"
  }
]
